source/minar_hal_timer.cpp

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (c) 2016 ARM Limited, All Rights Reserved
+ */
+
+// Include before mbed.h to properly get UINT*_C()
+
+#include "ns_types.h"
+
+#include "platform/arm_hal_timer.h"
+#include "platform/arm_hal_interrupt.h"
+
+#if defined(NS_EVENTLOOP_USE_TICK_TIMER) && defined(YOTTA_CFG)
```

Comment:
I think "NS_EVENTLOOP_USE_TICK_TIMER" could be implicit for mbed OS 3.

You could do

    #ífdef TARGET_LIKE_MBED && defined YOTTA_CFG
    #define NS_EVENTLOOP_USE_TICK_TIMER
    #endif

in the header file.

Or can you get NS_EVENTLOOP_USE_TICK_TIMER defined in a Yotta JSON file?